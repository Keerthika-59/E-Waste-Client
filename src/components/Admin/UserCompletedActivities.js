import React, { useEffect, useState, useMemo } from "react";
import Cookies from "js-cookie";
import APIHelper from "../API/apihelper";
import { Table } from "react-bootstrap";
const UserCompletedActivities = () => {
  const [compActivity, setCompActivity] = useState([]);
  const [id, setId] = useState("");

  const fetchId = async () => {
    try {
      const token = Cookies.get("user");
      const id = await APIHelper.fetchUserId({
        token: token,
      });
      setId(id);
    } catch (e) {
      console.log(e);
    }
  };

  useEffect(() => {
    fetchId();
  }, []);

  useEffect(() => {
    const fetchUserData = async () => {
      try {
        const userData = await APIHelper.fetchUserData(id);
        console.log(userData.activity);
        setCompActivity(userData.activity);
      } catch (err) {
        console.log(err.response || err);
      }
    };
    fetchUserData();
  }, [id]);

  return (
    <>
      <Table responsive striped bordered hover>
        <thead>
          <tr>
            <th>Activity Id</th>
            <th>Bio Waste</th>
            <th>Non Bio Waste</th>
            <th>Donation</th>
            <th>Representative Id</th>
          </tr>
        </thead>
        <tbody>
          {compActivity
            .filter((activity) => activity.status === true)
            .map((act) => (
              <tr>
                <td>{act._id}</td>
                <td>{act.bioWaste ? "Yes" : "No"}</td>
                <td>{act.nonBioWaste ? "Yes" : "No"}</td>
                <td>{act.donation ? "Yes" : "No"}</td>
                <td>{act.repId}</td>
              </tr>
            ))}
        </tbody>
      </Table>
    </>
  );
};

export default UserCompletedActivities;