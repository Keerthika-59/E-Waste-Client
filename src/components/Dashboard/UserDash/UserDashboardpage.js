import React,{useContext} from 'react'
import { Redirect } from "react-router";
import AuthApi from "../../../authAPI";
import Activity from './pages/activity';
function UserDashboard() {
    
    const Auth = useContext(AuthApi)
    
        return (
            (Auth.auth) ? (
            <>
                <Activity />
            </>):(<Redirect to='/UserLogIn' />)
        )
    }

    export default UserDashboard;