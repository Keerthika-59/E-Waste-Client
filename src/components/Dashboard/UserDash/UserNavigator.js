import React from 'react'
import { BrowserRouter as Route, Switch } from "react-router-dom";
import Activity from './pages/activity';
import Pending from './pages/pending';
import Completed from './pages/completed';

function Navigator() {
    return (
        <>
            <Switch>
                <Route path='/activity' exact component={Activity} />
                <Route path='/pending' exact component={Pending} />
                <Route path='/completed' exact component={Completed} />
            </Switch>
        </>
    )

}

export default Navigator;