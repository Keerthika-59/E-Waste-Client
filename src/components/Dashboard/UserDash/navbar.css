.sidenav {
  background-color: #ffd000e8;
  height: 40px;
  display: flex;
  justify-content: start;
  align-items: center;
}

.menu-bars {
    margin-left: 2rem;
    font-size: 1.5rem;
    background: none;
    
  }

.sidenav-menu {
  background-color: #004430;
  width: 270px;
  height: 100vh;
  display: flex;
  justify-content: center;
  position: relative;
  top: 100;
  left: -100%;
  transition: 850ms;
}

.sidenav-menu.active {
  left: 0;
  transition: 350ms;
}

.nav-text {
  display: flex;
  justify-content: start;
  align-items: center;
  padding: 18px 0px 8px 16px;
  list-style: none;
  height: 60px;
}

.nav-text  {
  text-decoration: none;
  color: #f5f5f5;
  font-size: 18px;
  display: flex;
  align-items: center;
  padding: 0 1px;
  border-radius: 4px;
  padding-top: 20px;
  padding-bottom: 20px;
}

.nav-text:hover {
  background-color: #546b85;
}

.sidenav-menu-items {
    top:20px;
  width: 100%;
}

.navbar-toggle {
  background-color: #060b26;
  width: 100%;
  height: 80px;
  display: flex;
  justify-content: start;
  align-items: center;
}

span {
  margin-left: 16px;
}

@media screen and (max-width: 767px) 
{
    .sidenav-menu {
        display: flex;
        flex-direction: column;
        width: 100%;
        height: 40vh;
        position: relative;
        top: 0px;
        left: -100%;
        opacity: 1;
        transition: all 0.5s ease;
    }
    .col-md-8
    {
        display: flex;
        flex-direction: column;
        width: 98%;
    }
}