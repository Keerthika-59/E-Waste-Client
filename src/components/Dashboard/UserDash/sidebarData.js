import React from 'react';

import * as BsIcons from 'react-icons/bs';
import Activity from './pages/activity';
import Pending from './pages/pending';

export const SidebarData = [
  {
    title: 'Create Activity',
    icon: <BsIcons.BsFillPersonPlusFill />,
    cName: 'nav-text'
  },
  
  {
    title: 'Pending Activities',
    icon: <BsIcons.BsFillPersonFill />,
    cName: 'nav-text'
  },
  {
    title: 'Completed Activities',
    icon: <BsIcons.BsFillPersonCheckFill />,
    cName: 'nav-text'
  },
];