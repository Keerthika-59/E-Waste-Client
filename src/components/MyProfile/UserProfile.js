import React, { useState, useEffect, useContext } from "react";
import Cookies from "js-cookie";
import axios from "axios";
import { Button } from "react-bootstrap";
import AuthApi from "../../authAPI";
import { Redirect } from "react-router";
import "./UserProfile.css";
// const phoneRegExp = /^[0-9]{10}$/g;
// const nameRegExp = /^[a-zA-Z ]{2,30}$/;

const UserProfile = () => {
  const Auth = useContext(AuthApi);
  const [edit, setEdit] = useState(false);
  const [token, setToken] = useState("");
  const [users, setUsers] = useState({
    // name: "",
    // phoneNumber: "",
    // email: "",
    // gender: "",
    // password: "",
    // city: "",
    // address: "",
  });
  const [updated, setUpdated] = useState({
    // name: "",
    // phoneNumber: "",
    // gender: "",
    // address: "",
  });

  const fetchId = async () => {
    try {
      const token = Cookies.get("user");
      const t = await axios.post("http://localhost:5000/auth1/getId", {
        token: token,
      });

      setToken(t.data);
    } catch (e) {
      console.log(e.message);
    }
  };

  useEffect(() => {
    fetchId();
  }, []);

  useEffect(() => {
    const fetchUserData = async () => {
      try {
        const userData = await axios.get(
          `http://localhost:5000/auth1/user/${token}`
        );
        setUsers(userData.data);
        setUpdated(userData.data);
      } catch (err) {
        console.log(err.response);
      }
    };
    fetchUserData();
  }, [token]);

  const handleChange = (e) => {
    const value = e.target.value;
    setUpdated({
      ...updated,
      [e.target.name]: value,
    });
  };

  const editClick = () => {
    console.log("clicked");
    setEdit(true);
  };
  const handleUpdate = async (e) => {
    e.preventDefault();
    console.log(updated);
    console.log(users);
    try {
      await axios.put(`http://localhost:5000/auth1/user/${token}`, updated);
      setEdit(false);
    } catch (err) {
      console.log(err);
    }
  };
  const handleCancel = () => {
    setUpdated(users);
    setEdit(false);
  };

  return Auth.auth ? (
    <div className="container rounded bg-white mt-5">
      <div className="row">
        <div className="col-md-4 border-right">
          <div className="d-flex flex-column align-items-center text-center p-3 py-5">
            <img
              className="rounded-circle mt-5"
              src="https://www.freeiconspng.com/thumbs/computer-user-icon/computer-user-icon-28.png"
              width="90"
              alt=""
            />
            <span className="font-weight-bold">{users.name}</span>
            <span className="text-black-50">{users.email}</span>
            <span>{users.city}</span>
          </div>
        </div>
        <div className="col-md-8">
          <div className="p-3 py-5">
            <div className="justify-content-between align-items-center mb-3">
              {/* <div className="d-flex flex-row align-items-center back"><i className="fa fa-long-arrow-left mr-1 mb-1"></i>
                                    <h6>Back to home</h6>
                                </div> */}
              <h4 className="text-center">My Profile</h4>
              <button onClick={editClick}>edit</button>
              <br />
            </div>

            <div className="row mt-2">
              <div className="col-md-6">
                <label>Name</label>
                <input
                  type="text"
                  className="form-control"
                  placeholder="Name"
                  name="name"
                  value={updated.name}
                  onChange={handleChange}
                  disabled={edit ? false : true}
                />
              </div>
              <div className="col-md-6">
                <label>Phone Number</label>
                <input
                  type="text"
                  className="form-control"
                  name="phoneNumber"
                  value={updated.phoneNumber}
                  onChange={handleChange}
                  placeholder="Phone Number"
                  disabled={edit ? false : true}
                />
              </div>
            </div>
            <div className="row mt-3">
              <div className="col-md-6">
                <label>Email</label>
                <input
                  type="text"
                  className="form-control"
                  placeholder="Email"
                  value={users.email}
                  disabled={true}
                />
              </div>
              <div className="col-md-6">
                <label>Gender</label>
                <input
                  type="text"
                  className="form-control"
                  value={users.gender}
                  placeholder="Gender"
                  disabled={true}
                />
              </div>
            </div>
            <div className="row mt-3">
              <div className="col-md-6">
                <label>Address</label>
                <input
                  type="text"
                  className="form-control"
                  placeholder="Address"
                  name="address"
                  onChange={handleChange}
                  value={updated.address}
                  disabled={edit ? false : true}
                />
              </div>
              <div className="col-md-6">
                <label>City</label>
                <input
                  type="text"
                  className="form-control"
                  value={users.city}
                  disabled={true}
                  placeholder="City"
                />
              </div>
            </div>
            {/* <div className="row mt-3">
                                <div className="col-md-6"><input type="text" className="form-control" placeholder="Bank Name" value="Bank of America"/></div>
                                <div className="col-md-6"><input type="text" className="form-control" value="043958409584095" placeholder="Account Number"/></div>
                            </div> */}

            <div className="mt-5 form-row">
              <Button
                variant="success"
                disabled={edit ? false : true}
                onClick={handleUpdate}
              >
                Update{" "}
              </Button>
              <pre> </pre>
              <Button
                variant="danger"
                disabled={edit ? false : true}
                onClick={handleCancel}
              >
                Cancel
              </Button>
            </div>
          </div>
        </div>
      </div>
    </div>
  ) : (
    <Redirect to="/UserLogIn" />
  );
};

export default UserProfile;
