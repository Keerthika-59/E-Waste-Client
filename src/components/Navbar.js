import React, { useState, useEffect } from 'react';
import { Button } from './Button';
import { Link } from 'react-router-dom';
import './Navbar.css';
import './Button.css'

function Navbar() {
  const [click, setClick] = useState(false);
  const [button, setButton] = useState(true);

  const handleClick = () => setClick(!click);
  const closeMobileMenu = () => setClick(false);

  const showButton = () => {
    if (window.innerWidth <= 960) {
      setButton(false);
    } else {
      setButton(true);
    }
  };

  useEffect(() => {
    showButton();
  }, []);

  window.addEventListener('resize', showButton);

  return (
    <>
      <nav className='navbar'>
        <div className='navbar-container'>
          <Link to='/' className='navbar-logo' onClick={closeMobileMenu}>
            <img src={`${process.env.PUBLIC_URL}/assets/images/wm-logo.png`} style={{height:"60px"}}></img>
          </Link>
          <div className='menu-icon' onClick={handleClick}>
            <i className={click ? 'fas fa-times' : 'fas fa-bars'} />
          </div>
          <ul className={click ? 'nav-menu active' : 'nav-menu'} style={{marginRight:"450px"}}>
            <li className='nav-item'>
              <Link to='/Gallery' className='nav-links' onClick={closeMobileMenu}>
                Gallery
              </Link>
            </li>
            <li className='nav-item'>
              <Link
                to='/AboutUs'
                className='nav-links'
                onClick={closeMobileMenu}
              >
                About Us
              </Link>
            </li>
            <li className='nav-item'>
              <Link
                to='/ContactUs'
                className='nav-links'
                onClick={closeMobileMenu}
              >
                Contact Us
              </Link>
            </li>

            <li>
              <Link
                to='/SignUp'
                className='nav-links-mobile'
                onClick={closeMobileMenu}
              >
                Sign Up
              </Link>
            </li>
            <li>
              <Link
                to='/LogIn'
                className='nav-links-mobile'
                onClick={closeMobileMenu}
              >
                Log In
              </Link>
            </li>
          </ul>
          <div style={{marginRight:"10px"}}>{button && <Button buttonStyle='btn--outline'>
            <Link to='/SignUP'>SIGN UP</Link></Button>}</div>
          <div>{button && <Button buttonStyle='btn--outline'><Link to='/LogIn'>LOG IN</Link></Button>}</div>
        </div>
      </nav>
    </>
  );
}

export default Navbar;
