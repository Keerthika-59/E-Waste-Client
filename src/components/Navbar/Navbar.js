import React, { useState, useEffect, useContext } from "react";
import { Link } from "react-router-dom";
import { NavDropdown,Dropdown, DropdownButton } from "react-bootstrap";
import "./Navbar.css";
import "../Button/Button.css";
import { Button } from "../Button/Button";
import Cookies from 'js-cookie'
import AuthApi from '../../authAPI'
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faSignOutAlt} from "@fortawesome/free-solid-svg-icons";
import { faUserCircle } from "@fortawesome/free-regular-svg-icons";
import { Redirect } from "react-router";

function Navbar() {

  const [click, setClick] = useState(false);
  const [button, setButton] = useState(true);
  const [userLog, setUser] = useState(false);
  const [repLog, setRep] = useState(false);  


  const handleClick = () => setClick(!click);
  const closeMobileMenu = () => setClick(false);

  const showButton = () => {
    if (window.innerWidth <= 1000) {
      setButton(false);
    } else {
      setButton(true);
    }
  };

  function readUserLog() {
    if (Cookies.get('user')) {
      setUser(true)
    }
    else {
      setUser(false)
    }
  }

  function readRepLog() {
    if (Cookies.get('repr')) {
      setRep(true)
    }
    else {
      setRep(false)
    }
  }


  useEffect(() => {
    const interval = setInterval(() => {
      readUserLog();
      readRepLog()
    }, 1);
    return () => clearInterval(interval);
  }, []);

  
  const Auth = useContext(AuthApi)
  const handleLogoutUser = async (e) => {
    e.preventDefault()


    Auth.setAuth(false)
    Cookies.remove('user')
    console.log("logged out");
    // props.history.push('/UserLogin')
    <Redirect to='/UserLogIn' />
  }

  const handleLogoutRep = async (e) => {
    e.preventDefault()


    Auth.setAuth(false)
    Cookies.remove('repr')
    console.log("logged out");
    <Redirect to='/RepresentativeLogIn' />
  }

  // console.log(userLog)
  useEffect(() => {
    showButton();
  }, []);


  window.addEventListener("resize", showButton);

  return (
    <>
      <nav className="navbar">
        <div className="navbar-container mr-auto">
          <Link
            to="/"
            className="navbar-logo ml-auto"
            onClick={closeMobileMenu}
          >
            <img
              src={`${process.env.PUBLIC_URL}/assets/images/wm-logo.png`}
              alt=""
              style={{ height: "70px" }}
            />
          </Link>
          <div className="menu-icon" onClick={handleClick}>
            <i className={click ? "fas fa-times" : "fas fa-bars"} />
          </div>
          <ul
            className={
              click ? "nav-menu active my-4 px-4" : "nav-menu my-4 px-4"
            }
          >
            <li className="nav-item">
              <Link
                to="/Gallery"
                style={{ textDecoration: "none" }}
                className="nav-links"
                onClick={closeMobileMenu}
              >
                Gallery
              </Link>
            </li>

      
      <li className="nav-item">
      <NavDropdown title="About Us" className="nav-links" id="basic-nav-dropdown">
      <NavDropdown.Item>
                 <Link
                      to="/AboutUs"
                      // className="nav-links"
                      onClick={closeMobileMenu}
                      style={{ textDecoration: "none" }}
                    >
                       <div>About Us </div>
                     
                </Link>
                </NavDropdown.Item>
                
              <NavDropdown.Item>
                <Link
                      to="/Goals"
                      // className="nav-links"
                      onClick={closeMobileMenu}
                      style={{ textDecoration: "none" }}
                    >
                       <div>Goals</div>
                     
                </Link>
              </NavDropdown.Item>
            
              <NavDropdown.Item>
                <Link
                      to="/Achievements"
                      // className="nav-links"
                      onClick={closeMobileMenu}
                      style={{ textDecoration: "none" }}
                    >
                       <div>Achievements</div>
                      
                </Link>
              </NavDropdown.Item>

              <NavDropdown.Item>
                <Link
                      to="/OurTeam"
                      // className="nav-links"
                      onClick={closeMobileMenu}
                      style={{ textDecoration: "none" }}
                    >
                     
                      <div>Our Team</div>
                </Link>
              </NavDropdown.Item>

            </NavDropdown>
                
            </li>

            <li className="nav-item">
      <NavDropdown title="Services" className="nav-links" id="basic-nav-dropdown">
             
             <NavDropdown.Item>
                <Link
                      to="/Services"
                      // className="nav-links"
                      onClick={closeMobileMenu}
                      style={{ textDecoration: "none" }}
                    >
                      
                      <div>Our Services</div>
                </Link>
              </NavDropdown.Item>
              <NavDropdown.Item>
                <Link
                      to="/Ewaste"
                      // className="nav-links"
                      onClick={closeMobileMenu}
                      style={{ textDecoration: "none" }}
                    >
                       <div>E-Waste</div>
                     
                </Link>
              </NavDropdown.Item>
            
              <NavDropdown.Item>
                <Link
                      to="/Disposing"
                      // className="nav-links"
                      onClick={closeMobileMenu}
                      style={{ textDecoration: "none" }}
                    >
                        <div>Dispose</div>
                      
                </Link>
              </NavDropdown.Item>

              <NavDropdown.Item>
                <Link
                      to="/Donating"
                      // className="nav-links"
                      onClick={closeMobileMenu}
                      style={{ textDecoration: "none" }}
                    >
                     <div>
                    Donate
                  </div>
                </Link>
              </NavDropdown.Item>

            </NavDropdown>
                
            </li>
            <li className="nav-item">
                <Link
                      to="/ContactUs"
                      className="nav-links"
                      onClick={closeMobileMenu}
                      style={{ textDecoration: "none",width:"130px" }}
                    >
                    Contact Us
                    </Link>
            </li>
            <li>
              <Link
                to="/SignUpoptions"
                className="nav-links-mobile"
                onClick={closeMobileMenu}
                style={{ textDecoration: "none" }}
              >
                Signup
              </Link>
            </li>
            <li>
              <Link
                to="/LogInoptions"
                className="nav-links-mobile"
                // ref="loginbtn"
                onClick={closeMobileMenu}
                style={{ textDecoration: "none" }}
              >
                Log In
              </Link>
            </li>
          </ul>
        </div>

        {/* <div className="login-navbar mr-auto "> */}
          {button && (!userLog) && (!repLog) && (
            <DropdownButton
              className="mr-auto"
              id="dropdown-basic-button"
              title="LOG IN"
            >
              <Dropdown.Item>
                <Link to="/UserLogIn"
                style={{ textDecoration: "none" }}>
                  <div>
                   User
                  </div>
                </Link>
              </Dropdown.Item>
              <Dropdown.Divider />
              <Dropdown.Item>
                <Link to="/RepresentativeLogIn"
                 style={{ textDecoration: "none" }}>
                  <div>
                  Representative
                  </div>
                </Link>
              </Dropdown.Item>
            </DropdownButton>
          )}
        {/* </div> */}

        {/* <div className="login-navbar mr-auto "> */}
          {button && (userLog) && (
            <DropdownButton
              className="mr-auto"
              id="dropdown-basic-button"
              title="User"

            >
            

             <Dropdown.Item className="fw-bold">
              <Link to='/MyProfile' 
                style={{ textDecoration: "none" }}
                >
                  <div>
                <FontAwesomeIcon icon={faUserCircle} className="me-2" /> 
                My Profile
                </div>
                </Link>
             </Dropdown.Item>

              <Dropdown.Divider />

              <Dropdown.Item className="fw-bold">
              <Link to='/UserDash' 
                style={{ textDecoration: "none" }}
                >
                <div>
                <FontAwesomeIcon icon={faUserCircle} className="me-2" /> Dashboard
                </div>
                </Link>
             </Dropdown.Item>

              {/* <Dropdown.Divider />

              <Dropdown.Item className="fw-bold">
              <Link to='/ActivityDetails' 
                style={{ textDecoration: "none" }}
                >
                <div>
                <FontAwesomeIcon icon={faUserCircle} className="me-2" /> Activity
                </div>
                </Link>
             </Dropdown.Item> */}

              <Dropdown.Divider />
              

              <Dropdown.Item className="fw-bold" onClick={e => handleLogoutUser(e)}>
               <div>
                <FontAwesomeIcon icon={faSignOutAlt} className="text-danger me-2" /> Logout
                </div>
             </Dropdown.Item>
            </DropdownButton>
          )}
        {/* </div> */}

        {/* <div className="login-navbar mr-auto "> */}
          {button && (repLog)&& (
            <DropdownButton
              className="mr-auto"
              id="dropdown-basic-button"
              title="Representative "
            >
              <Dropdown.Item className="fw-bold">
              <Link to='/RepProfile' 
                style={{ textDecoration: "none" }}
                >
                <FontAwesomeIcon icon={faUserCircle} className="me-2" /> My Profile</Link>
             </Dropdown.Item>
              <Dropdown.Divider />

              <Dropdown.Item className="fw-bold">
              <Link to='/RepDash' 
                style={{ textDecoration: "none" }}
                >
                <FontAwesomeIcon icon={faUserCircle} className="me-2" /> Dashboard</Link>
             </Dropdown.Item>



              <Dropdown.Divider />

              <Dropdown.Item className="fw-bold" onClick={e => handleLogoutRep(e)}>
                <FontAwesomeIcon icon={faSignOutAlt} className="text-danger me-2" /> Logout
             </Dropdown.Item>
            </DropdownButton>
          )}
        {/* </div> */}
      </nav>
    </>
  );
}

export default Navbar;
