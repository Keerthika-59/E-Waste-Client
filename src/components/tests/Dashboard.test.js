import React from "react";
import { shallow } from 'enzyme';
import ReactDOM from "react-dom";
import { Route } from "react-router-dom";
import Completed from "../Dashboard/RepDash/pages/repCompleted";
import Activity from "../Dashboard/UserDash/pages/activity";
import UserCompletedActivities from "../Dashboard/UserDash/pages/completed";
import Pending from "../Dashboard/UserDash/pages/pending";
import UserProfile from "../Dashboard/UserDash/pages/Profile/userProfile";
import UserDashboard from "../Dashboard/UserDash/UserDashboardpage";
import UserDash from "../Dashboard/UserDash/Usericon";
import Navbar from "../Dashboard/UserDash/UserSidebar";
import PendingR from "../Dashboard/RepDash/pages/repPending"
import RepProfile from "../Dashboard/RepDash/Profile/repProfile";
import RepNavbar from "../Dashboard/RepDash/RepSidebar";
import RepDashboard from "../Dashboard/RepDash/RepDashboardpage";

test("renders representative dashboard", () => {
    const div = document.createElement("div");
    <Route>
      ReactDom.render(
      <RepDashboard />
      ,div);
    </Route>;
    ReactDOM.unmountComponentAtNode(div);
  });
  // test('should test RepDashboard component', () => {
  //   const wrapper = shallow(<RepDashboard />);
  //   expect(wrapper).toMatchSnapshot();
  //  });
test("renders representative sidebar", () => {
    const div = document.createElement("div");
    <Route>
      ReactDom.render(
      <RepNavbar />
      ,div);
    </Route>;
    ReactDOM.unmountComponentAtNode(div);
  });
  // test('should test RepNavbar component', () => {
  //   const wrapper = shallow(<RepNavbar />);
  //   expect(wrapper).toMatchSnapshot();
  //  });
test("renders representative profile", () => {
    const div = document.createElement("div");
    <Route>
      ReactDom.render(
      <RepProfile />
      ,div);
    </Route>;
    ReactDOM.unmountComponentAtNode(div);
  });
  // test('should test RepProfile component', () => {
  //   const wrapper = shallow(<RepProfile />);
  //   expect(wrapper).toMatchSnapshot();
  //  });
test("renders representative Pending activities", () => {
    const div = document.createElement("div");
    <Route>
      ReactDom.render(
      <PendingR />
      ,div);
    </Route>;
    ReactDOM.unmountComponentAtNode(div);
  });
  test('should test repPending component', () => {
    const wrapper = shallow(<PendingR />);
    expect(wrapper).toMatchSnapshot();
   });
test("renders representative's completed activities", () => {
    const div = document.createElement("div");
    <Route>
      ReactDom.render(
      <Completed />
      ,div);
    </Route>;
    ReactDOM.unmountComponentAtNode(div);
  });
  test('should test repCompleted component', () => {
    const wrapper = shallow(<Completed />);
    expect(wrapper).toMatchSnapshot();
   });
test("renders User sidebar", () => {
    const div = document.createElement("div");
    <Route>
      ReactDom.render(
      <Navbar />
      ,div);
    </Route>;
    ReactDOM.unmountComponentAtNode(div);
  });
  // test('should test user Navbar component', () => {
  //   const wrapper = shallow(<Navbar />);
  //   expect(wrapper).toMatchSnapshot();
  //  });
test("renders User Navbar", () => {
    const div = document.createElement("div");
    <Route>
      ReactDom.render(
      <UserDash />
      ,div);
    </Route>;
    ReactDOM.unmountComponentAtNode(div);
  });
  // test('should test UserDash component', () => {
  //   const wrapper = shallow(<UserDash />);
  //   expect(wrapper).toMatchSnapshot();
  //  });
test("renders User Dashboard", () => {
    const div = document.createElement("div");
    <Route>
      ReactDom.render(
      <UserDashboard />
      ,div);
    </Route>;
    ReactDOM.unmountComponentAtNode(div);
  });
  // test('should test UserDashboard component', () => {
  //   const wrapper = shallow(<UserDashboard />);
  //   expect(wrapper).toMatchSnapshot();
  //  });
test("renders user Pending activities", () => {
    const div = document.createElement("div");
    <Route>
      ReactDom.render(
      <Pending />
      ,div);
    </Route>;
    ReactDOM.unmountComponentAtNode(div);
  });
  test('should test UserPending component', () => {
    const wrapper = shallow(<Pending />);
    expect(wrapper).toMatchSnapshot();
   });
test("renders user completed activities", () => {
    const div = document.createElement("div");
    <Route>
      ReactDom.render(
      <UserCompletedActivities />
      ,div);
    </Route>;
    ReactDOM.unmountComponentAtNode(div);
  });
  test('should test UserCompletedActivities component', () => {
    const wrapper = shallow(<UserCompletedActivities />);
    expect(wrapper).toMatchSnapshot();
   });
test("renders user profile", () => {
    const div = document.createElement("div");
    <Route>
      ReactDom.render(
      <UserProfile />
      ,div);
    </Route>;
    ReactDOM.unmountComponentAtNode(div);
  });
  // test('should test UserProfile component', () => {
  //   const wrapper = shallow(<UserProfile />);
  //   expect(wrapper).toMatchSnapshot();
  //  });
test("renders create user activity", () => {
    const div = document.createElement("div");
    <Route>
      ReactDom.render(
      <Activity />
      ,div);
    </Route>;
    ReactDOM.unmountComponentAtNode(div);
  });
  test('should test Activity component', () => {
    const wrapper = shallow(<Activity />);
    expect(wrapper).toMatchSnapshot();
   });