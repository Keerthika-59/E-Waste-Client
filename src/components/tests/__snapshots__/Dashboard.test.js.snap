// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test Activity component 1`] = `
<div
  class="container"
>
  <div
    class="row"
  >
    <div
      class="holder"
    >
      <form
        className="form-content"
        onSubmit={[Function]}
      >
        <table
          width="100%"
        >
          <thead>
            <th
              class="actHeading"
            >
              User activity
            </th>
          </thead>
          <tr>
            <td>
              Biodegradable
            </td>
            <td>
              <div
                class="outerDivFull"
              >
                <div
                  class="switchToggle"
                >
                  <input
                    id="switch"
                    onChange={[Function]}
                    type="checkbox"
                    value={false}
                  />
                  <label
                    for="switch"
                  />
                </div>
              </div>
            </td>
          </tr>
          <tr>
            <td>
              Non-Biodegradable
            </td>
            <td>
              <div
                class="outerDivFull"
              >
                <div
                  class="switchToggle"
                >
                  <input
                    id="switch1"
                    onChange={[Function]}
                    type="checkbox"
                    value={false}
                  />
                  <label
                    for="switch1"
                  />
                </div>
              </div>
            </td>
          </tr>
          <tr>
            <td
              class="donation"
            >
              Donation
            </td>
            <td>
              <div
                class="outerDivFull"
              >
                <div
                  class="switchToggle"
                >
                  <input
                    id="switch2"
                    onChange={[Function]}
                    type="checkbox"
                    value={false}
                  />
                  <label
                    for="switch2"
                  />
                </div>
              </div>
            </td>
          </tr>
          <br />
          <tr>
            <td
              colspan="2"
            >
              <input
                disabled={true}
                onSubmit={[Function]}
                type="submit"
                value="Submit"
              />
            </td>
          </tr>
        </table>
      </form>
    </div>
  </div>
</div>
`;

exports[`should test UserCompletedActivities component 1`] = `
<Fragment>
  <ForwardRef
    bordered={true}
    hover={true}
    striped={true}
  >
    <thead>
      <tr>
        <th>
          Activity Id
        </th>
        <th>
          Bio Waste
        </th>
        <th>
          Non Bio Waste
        </th>
        <th>
          Donation
        </th>
        <th>
          Representative Id
        </th>
      </tr>
    </thead>
    <tbody>
      <div>
        Loading...
      </div>
    </tbody>
  </ForwardRef>
</Fragment>
`;

exports[`should test UserPending component 1`] = `
<Fragment>
  <h3
    className="pendingHeading"
  >
    Pending activity
  </h3>
  <div>
    Loading...
  </div>
</Fragment>
`;

exports[`should test repCompleted component 1`] = `
<Fragment>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="col-md-10"
      >
        <div
          class="card  px-3 py-4 my-4 mx-5 justify-content-center"
        >
          <div
            class="card-body1"
          >
            <h6
              class="card-title"
            >
              Timeline
            </h6>
            <div
              id="content"
            >
              <ul
                class="timeline"
              >
                <li
                  class="event"
                  data-date="22/03/2021"
                >
                  <h3>
                    Activity ID
                  </h3>
                  <p>
                    Representative name
                  </p>
                </li>
                <li
                  class="event"
                  data-date="2:30 - 4:00pm"
                >
                  <h3>
                    Activity ID
                  </h3>
                  <p>
                    Representative name
                  </p>
                </li>
                <li
                  class="event"
                  data-date="5:00 - 8:00pm"
                >
                  <h3>
                    Activity ID
                  </h3>
                  <p>
                    Representative name
                  </p>
                </li>
                <li
                  class="event"
                  data-date="8:30 - 9:30pm"
                >
                  <h3>
                    Activity ID
                  </h3>
                  <p>
                    Representative name
                  </p>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</Fragment>
`;

exports[`should test repPending component 1`] = `
<Fragment>
  <h3
    className="pendingHeading"
  >
    Pending activity
  </h3>
  <div>
    Loading...
  </div>
</Fragment>
`;
